RFS/src/main/java/com/rfs/ReindexFromSnapshot.java

```diff
@@ -358,12 +361,17 @@ public static void main(String[] args) throws InterruptedException {
 
                         Flux<Document> documents = new LuceneDocumentsReader().readDocuments(luceneDirPath, indexMetadata.getName(), shardId);
                         String targetIndex = indexMetadata.getName() + indexSuffix;
-                        DocumentReindexer.reindex(targetIndex, documents, targetConnection);
 
-                        logger.info("Shard reindexing completed");
+                        int targetShardId = shardId; // Define in local context for the lambda
```

Comment:
super minor style thing - you can make this explicitly final and keep the same name other than some convention (like prepending the word 'final') to show that you're only doing it for the lambda.

Reply:
Thanks - done.